storage/ctladdr/multiaddresses.go

```diff
@@ -0,0 +1,76 @@
+package ctladdr
+
+import (
+	"context"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+type MultiAddressSelector struct {
```

Comment:
This is lotus-provider only right? In that case I'd move it to `provider/[address.go]`